Neos.ContentRepositoryRegistry/Classes/Command/SubprocessProjectionCatchUpCommandController.php

```diff
@@ -31,7 +40,10 @@ class SubprocessProjectionCatchUpCommandController extends CommandController
      */
     public function catchupCommand(string $contentRepositoryIdentifier, string $projectionClassName): void
     {
-        $contentRepository = $this->contentRepositoryRegistry->get(ContentRepositoryId::fromString($contentRepositoryIdentifier));
+        $contentRepositoryId = ContentRepositoryId::fromString($contentRepositoryIdentifier);
+        $runnerState = AsynchronousCatchUpRunnerState::create($contentRepositoryId, $projectionClassName, $this->catchUpStatesCache);
```

Comment:
Wouldn't it make sense to make `AsynchronousCatchUpRunnerState` a singleton that already has the cache backend injected.
Otherwise we leak internal details and the exact cache name has to be specified everywhere this is used (and using a different cache will work, but not as expected).
Of course, as a consequence `$contentRepositoryId` and `$projectionClassName` would have to be arguments of the public methods then..
Alternatively we could create a factory and inject _that_:
```php
$this->catchUpRunnerStateFactory->build($contentRepositoryId, $projectionClassName)->setStopped();
```

nit pick: the `$runnerState` instance does not have to be created (and occupy memory) before the catchup is started